### ISSUE TITLE
clarify WSL support policy

### ISSUE THREAD

**Issue by mknyszek** [golang/go#63503]
In triage we were discussing #63329 and we weren't actually sure what the support policy is for WSL1 vs. WSL2. Looking at https://github.com/golang/go/wiki/PortingPolicy, I don't see any concrete statement about this. It is a somewhat unique platform, so it would be good to identify what our support policy is and how we should handle such issues.

CC @golang/windows

**Comment by bcmills** [comment]
It's not an explicit policy, but in general for configurations for which we don't have a builder (WSL1, unusual `git` versions, unusual container environments, QEMU, etc.) we have tended to accept patches for simple/straightforward fixes, but not actively fix them ourselves (since we don't have a way to test a potential fix) and not accept complicated workarounds (since we don't have a way to run regression tests to verify that the workaround continues to work).

**Comment by qmuntal** [comment]
I've discussed about this issue with my team at Microsoft, and we might be interested on keeping WSL1 supported a bit more. We will evaluate if there is a need for that to happen and also how can we help supporting it.

**Comment by gdams** [comment]
Hi all 👋🏼 I'm the Go Group manager at Microsoft. We've been discussing this internally and checking usage with customers. One question we had was whether it would be possible to add a question to the next Go quarterly survey about WSL1 usage.

We know from the [previous survey](https://go.dev/blog/survey2023-q1-results) that ~15% of those who answered were developing on WSL but this didn't specifically ask the question as to which version of WSL they were using.

**Comment by ianlancetaylor** [comment]
CC @alicemerrick @fflewddur see previous comment

**Comment by alicemerrick** [comment]
That is something we could add as a follow up on the survey if there’s room. What’s your criteria for determining whether to support WSL1 ? For example, if more than x% of WSL users are using WSL1, we will support it a bit more.

**Comment by gdams** [comment]
> That is something we could add as a follow up on the survey if there’s room. What’s your criteria for determining whether to support WSL1 ? For example, if more than x% of WSL users are using WSL1, we will support it a bit more.

Yeah we'd like to better understand how many people in the community are still using WSL1. That number will help us determine a path forwards supporting it

**Comment by fflewddur** [comment]
Hi @gdams! I wanted to let you know this question was included in our latest survey. The survey is still in the field, but we have early results: currently we see 93% of respondents who use WSL saying they use WSL v2, 1% use WSL v1, 2% say they use both, and 4% aren't sure. 

This is from 426 developers who work with WSL, but given the lopsided distribution and our randomized sampling, I don't expect these numbers to meaningfully shift over the next two weeks. We'll share the final numbers once the survey closes on Feb 11.

**Comment by gdams** [comment]
Hi @fflewddur,

Thanks very much for sharing your findings here, based on our internal findings, as well as the ones from the community survey, we (the Go team at Microsoft) don't see any reason to provide additional support WSL1. We will be focussing on WSL2 going forwards.
